pkg/scaling/resolver/hashicorpvault_handler.go

```diff
@@ -110,6 +110,10 @@ func (vh *HashicorpVaultHandler) token(client *vaultapi.Client) (string, error)
 			return token, errors.New("k8s role not in config")
 		}
 
+		if vh.vault.Credential == nil {
```

Comment:
Could you please add a test case for this?

Reply:
I have added a test for this case